lib/rpc/server/peers.go

```diff
@@ -227,6 +227,7 @@ func (r *peers) reconnectPeer(p Peer, reqCh <-chan chan clientUpdate, doneCh cha
 			err := utils.RetryWithInterval(r.ctx, r.Backoff(), func() (err error) {
 				clt, err = p.Reconnect(r.ctx)
 				if err != nil {
+					log.WithError(err).Warn("Unable to reconnect.")
```

Comment:
Nit: to me unable to reconnect I think implies a permanent failure, IE I get the impression from the wording that the connection won't necessarily work in the future. Which presumably is what Should reconnect is trying to determine after this.

So I think it might be a bit clearer to indicate whether a reconnection will be attempted. Ie: "Failed to connect. Retry in 6s", probably with feedback from whether the code is actually treating this as a permanent failure or not.

Reply:
The reconnect attempts are already being logged with the corresponding amount of backoff. I'll remove this entry since it's superfluous.